web/session.go

```diff
@@ -83,6 +85,14 @@ func purgeOldsessionTs() {
 			delete(sessions, key)
 		}
 	}
+
+	for {
+		time.Sleep(time.Second * 60)
+
+		sessionsLock.Lock()
```

Comment:
I found a bug in the original impl where it failed to Unlock if there were 0 sessions. So if you start, wait > 1 minute, then you can't login. So I could have fixed that bug with an Unlock call before the continue, but I thought that left the routine vulnerable to some future bug if another continue was added.  I can move the Lock/Unlock into the closure if you think that's better.